### PR TITLE
[OSD-10340] update the status for managed notification object

### PR DESCRIPTION
**Pull request by bmeng**
### What type of PR is this?
_feature_

### What this PR does / why we need it?
Update the managednotification status based on the status transition

### Which Jira/Github issue(s) this PR fixes?

_Fixes #OSD-10340_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

**Comment by bmeng** [comment]
/hold
need rebase after https://github.com/openshift/ocm-agent-operator/pull/31 merged

**Comment by bmeng** [comment]
Code rebased and test on staging cluster.

Scenarios: 
1. When the alert first fired
```
  status:
    notifications:
    - conditions:
      - lastTransitionTime: "2022-03-16T09:19:44Z"
        reason: Alert is firing
        status: "True"
        type: AlertFiring
      - lastTransitionTime: "2022-03-16T09:19:44Z"
        reason: Alert resolved
        status: "False"
        type: AlertResolved
      - lastTransitionTime: "2022-03-16T09:19:44Z"
        reason: Service log sent
        status: "True"
        type: ServiceLogSent
      name: TestTemplate
      serviceLogSentCount: 1
```

2. When the resend window passed
```
  status:
    notifications:
    - conditions:
      - lastTransitionTime: "2022-03-16T09:19:44Z"
        reason: Alert is firing
        status: "True"
        type: AlertFiring
      - lastTransitionTime: "2022-03-16T09:19:44Z"
        reason: Alert resolved
        status: "False"
        type: AlertResolved
      - lastTransitionTime: "2022-03-16T09:34:45Z"
        reason: Service log sent
        status: "True"
        type: ServiceLogSent
      name: TestTemplate
      serviceLogSentCount: 2

```

The alert resolved can be handled by the status update, can be used directly when we implement the alert resolving in the `processAlert()`

File: pkg/handlers/webhookreceiver.go [hunk 1]
```diff
@@ -250,18 +253,52 @@ func (h *WebhookReceiverHandler) updateNotificationStatus(n *oav1alpha1.Notifica
...
+					status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
+				} else {
+					// Status transition is Firing to Resolved
+					// Update the condition status and timestamp for AlertFiring
+					// Update the condition status and timestamp for AlertResolved
+					// Update the timestamp for the ServiceLogSent
+					status.SetStatus(oav1alpha1.ConditionAlertFiring, "Alert is firing", corev1.ConditionFalse, timeNow)
+					status.SetStatus(oav1alpha1.ConditionAlertResolved, "Alert resolved", corev1.ConditionTrue, timeNow)
+					status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
+				}
```
Comment (ravitri) [inline pkg/handlers/webhookreceiver.go:290]: So from the flow I see that `processAMReceiver` method would called `processAlert` method in a loop for only the firing alerts. This is why, the newly added bool value for `updateNotificationStatus` is set to `true` always. Considering that, will this highlighted code in the `else` block be ever reached?

Comment (mrbarge) [reply pkg/handlers/webhookreceiver.go:290]: My guess is this is being written in advance of OSD-10225 to implement `resolved` handling (which would make the same call, but with `false`), but Meng Bo will be in the best position to advise.

Comment (bmeng) [reply pkg/handlers/webhookreceiver.go:290]: Yes, the block for the `!firing` is used for the resolve part in OSD-10225.
I am just thinking consider them together will be easier from the logic over all.

And when we are going to implement the resolve alert logic.
We can just call the same function `updateNotificationStatus()` with a `false` in the end, just like Matt mentioned.

We may need to do some small tweak in `processAlert()` function then if we want to use it in the loop `for _, alert := range d.Alerts.Resolved()` like what we are doing now in L107 for firing alerts.

**Pull request merged**